server/stt_metrics_test.go

```diff
@@ -0,0 +1,76 @@
+package server
+
+import (
+	"context"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/api"
+	promapi "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	pmmapitests "github.com/Percona-Lab/pmm-api-tests"
+)
+
+func TestSTTMetrics(t *testing.T) {
+	t.Run("StartSTTChecksAndRecordMetrics", func(t *testing.T) {
+		client, err := api.NewClient(api.Config{
+			Address: pmmapitests.BaseURL.ResolveReference(&url.URL{
+				Path: "/prometheus",
+			}).String(),
+		})
+		require.NoError(t, err)
+		promClient := promapi.NewAPI(client)
+
+		testCases := []struct {
+			query          string
+			len            int
+			metricType     string
+			expectedValues []string
+		}{
+			{
+				query:      "pmm_managed_checks_alerts_generated_total",
+				len:        7,
+				metricType: "vector",
+				expectedValues: []string{
+					`pmm_managed_checks_alerts_generated_total{check_type="MONGODB_BUILDINFO", instance="pmm-server", job="pmm-managed", service_type="mongodb"} => 0`,
+					`pmm_managed_checks_alerts_generated_total{check_type="MONGODB_GETCMDLINEOPTS", instance="pmm-server", job="pmm-managed", service_type="mongodb"} => 0`,
+					`pmm_managed_checks_alerts_generated_total{check_type="MONGODB_GETPARAMETER", instance="pmm-server", job="pmm-managed", service_type="mongodb"} => 0`,
+					`pmm_managed_checks_alerts_generated_total{check_type="MYSQL_SELECT", instance="pmm-server", job="pmm-managed", service_type="mysql"} => 0`,
+					`pmm_managed_checks_alerts_generated_total{check_type="MYSQL_SHOW", instance="pmm-server", job="pmm-managed", service_type="mysql"} => 0`,
+					`pmm_managed_checks_alerts_generated_total{check_type="POSTGRESQL_SELECT", instance="pmm-server", job="pmm-managed", service_type="postgresql"} => 0`,
+					`pmm_managed_checks_alerts_generated_total{check_type="POSTGRESQL_SHOW", instance="pmm-server", job="pmm-managed", service_type="postgresql"} => 0`},
+			},
+			{
+				query:      "pmm_managed_checks_scripts_executed_total",
+				len:        3,
+				metricType: "vector",
+				expectedValues: []string{
+					`pmm_managed_checks_scripts_executed_total{instance="pmm-server", job="pmm-managed", service_type="mongodb"} => 0`,
+					`pmm_managed_checks_scripts_executed_total{instance="pmm-server", job="pmm-managed", service_type="mysql"} => 0`,
+					`pmm_managed_checks_scripts_executed_total{instance="pmm-server", job="pmm-managed", service_type="postgresql"} => 0`},
+			},
+		}
+
+		for _, tc := range testCases {
+			result, _, err := promClient.Query(context.Background(),
+				tc.query, time.Now())
+
+			var actualValues []string
+			for _, s := range strings.Split(result.String(), "\n") {
```

Comment:
Why can't we loop over result? As far as I see it's just a slice of *models.Sample.

Reply:
could loop over the result but even then I will have to convert the result to a string inside the loop and remove the timestamp from it

Reply:
Checked structure of *models.Sample and it looks overcomplicated, so let's keep it.